### PR TITLE
allocate host blob from hw context

### PR DESCRIPTION
**Pull request by songbell**
Signed-off-by: fishbell <bell.song@intel.com>

### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/auto/executable_network.cpp [hunk 1]
```diff
@@ -553,16 +553,15 @@ InferenceEngine::IInferRequestInternal::Ptr MultiDeviceExecutableNetwork::Create
...
+    InferenceEngine::RemoteContext::Ptr ctx = nullptr;
 
     if (_workModeIsAUTO) {
         if (!_loadContext[CPU].isEnabled && _loadContext[ACTUALDEVICE].isAlready) {
-            auto& dev_requests = _workerRequests[_loadContext[ACTUALDEVICE].deviceInfo.deviceName];
-            if (num < dev_requests.size()) {
-                request_to_share_blobs_with = dev_requests.at(num)._inferRequest;
-            }
+            if (_loadContext[ACTUALDEVICE].deviceInfo.deviceName.find("CPU") == std::string::npos)
+                ctx = GetCore()->GetDefaultContext(_loadContext[ACTUALDEVICE].deviceInfo.deviceName);
```
Comment (ilya-lavrenov) [inline src/plugins/auto/executable_network.cpp:561]: can we remove this check `if` and use `try / catch` to check whether this method is implemented?

File: src/inference/src/ie_core.cpp [hunk 2]
```diff
@@ -497,6 +497,11 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
         return newAPI;
     }
 
+    ie::RemoteContext::Ptr GetDefaultContext(const std::string& deviceName) override {
+        auto parsed = ov::parseDeviceNameIntoConfig(deviceName, {});
+        return GetCPPPluginByName(parsed._deviceName).get_default_context(parsed._config)._ptr;
+    }
```
Comment (ilya-lavrenov) [inline src/inference/src/ie_core.cpp:503]: can we re-use this method / implementation in `Core::GetDefaultContext` ?

File: src/plugins/auto/executable_network.cpp (hunk 1 above)
Comment (songbell) [reply src/plugins/auto/executable_network.cpp]: thanks for suggestion, fixed.

File: src/inference/src/ie_core.cpp (hunk 2 above)
Comment (songbell) [reply src/inference/src/ie_core.cpp:503]: Hi ilya @ilya-lavrenov , we need to call the get context API in plugin, icore is the interface I can use.

Comment (ilya-lavrenov) [reply src/inference/src/ie_core.cpp:487]: I understand, but from Core we can also use ICore, but now we have 2 different implementations. Let's reuse ICore from Core

Comment (songbell) [reply src/inference/src/ie_core.cpp:487]: change the Core implementation, to reuse icore function.

**Review by myshevts — APPROVED** [review]

**Review by wangleis — APPROVED** [review]

**Pull request merged**